core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -588,7 +588,7 @@ import scala.util.control.NonFatal
               progressTracker.committed(offsets)
               replyTo.foreach(_ ! Done)
 
-            case e: RebalanceInProgressException => retryCommits(duration, e)
+            case e @ (_: RebalanceInProgressException | _: TimeoutException) => retryCommits(duration, e)
```

Comment:
I agree with your comment in https://github.com/akka/alpakka-kafka/issues/266#issuecomment-1320935836 that this could handle all kinds of `RetriableException` here instead. That would even catch `RetriableCommitFailedException` below. 
I assume most of the subclasses to `RetriableException` can not happen just here.

Reply:
@ennru I'll fix it and commit. I'll also take a look at the classes that (currently) extend `RetriableException`. But it should be safe to retry all of them inside a callback on the commit result (as this means they must have been thrown during the commit itself)

Reply:
@ennru The following (specific) exceptions will be handled if we decide to go for `RetriableException`:

1. CoordinatorLoadInProgressException
2. UnknownTopicOrPartitionException
3. CoordinatorNotAvailableException
4. NotCoordinatorException
5. TimeoutException

While it makes sense to retry a timed-out operation and the `CoordinatorLoadInProgressException` are clearly safe to retry (we've seen `CoordinatorLoadInProgressException` in our services a couple of times), the 3 other are not entirely clear to me, for instance: the `UnknownTopicOrPartitionException` comment states that 
```
This exception is used in contexts where a topic doesn't seem to exist based on possibly stale metadata.
 This exception is retriable because the topic or partition might subsequently be created.
```
Which feels that this operation might be retried endlessly (as currently `retryCommits()` doesn't have a limit on the number of retries, if I understand the code correctly)

Reply:
Thank you for looking closely at those sub-exceptions. It was the endless retrying I was afraid of, stale metadata sounds like a reason to try, but if the topic or partition is never created we it will loop.

Reply:
@ennru Thanks. Anything else on my side?